inc/Widget/SmailyWidget.php

```diff
@@ -63,7 +63,10 @@ public function widget( $args, $instance ) {
 		}
 
 		// Get current url.
-		$current_url = ( isset( $_SERVER['HTTPS'] ) ? 'https' : 'http' ) . '://' . $_SERVER['HTTP_HOST'] . $_SERVER['REQUEST_URI'];
+		$current_host = ( isset( $_SERVER['HTTPS'] ) ? 'https' : 'http' ) . '://' . $_SERVER['HTTP_HOST'];
+		// Remove params from redirect links.
+		$current_path = explode( '?', $_SERVER['REQUEST_URI'] );
+		$current_url = $current_host . $current_path[0];
```

Comment:
I think this is too harsh way to remove `code` and `message` arguments from the query, because you might accidentally remove arguments that are necessary for the current page/post to display. Try using Wordpress' [remove_query_arg](https://developer.wordpress.org/reference/functions/remove_query_arg/) function instead.

Reply:
Yes, this approach was too harsh if I think of it. I have removed only code and message arguments from the return URL. https://github.com/sendsmaily/smaily-woocommerce-plugin/pull/43/commits/908114bfdd9b5c85ae83fb794fd137fc13ab4870